src/overlays/actors/ovl_Bg_Dy_Yoseizo/z_bg_dy_yoseizo.c

```diff
@@ -71,55 +64,978 @@ const ActorInit Bg_Dy_Yoseizo_InitVars = {
+        }
+    } else {
+        switch (this->fountainType) {
+            case FAIRY_UPGRADE_SPIN_ATTACK:
+                if (!(gSaveContext.magicAcquired) || BREG(2)) {
+                    // Spin Attack speed UP
+                    osSyncPrintf(VT_FGCOL(GREEN) " ☆☆☆☆☆ 回転切り速度ＵＰ ☆☆☆☆☆ \n" VT_RST, &gSaveContext);
+                    this->givingSpell = true;
+                    givingReward = true;
+                }
+                break;
+            case FAIRY_UPGRADE_DOUBLE_MAGIC:
+                if (!(gSaveContext.doubleMagic)) {
+                    // Magic Meter doubled
+                    osSyncPrintf(VT_FGCOL(YELLOW) " ☆☆☆☆☆ 魔法ゲージメーター倍増 ☆☆☆☆☆ \n" VT_RST, &gSaveContext);
+                    this->givingSpell = true;
+                    givingReward = true;
+                }
+                break;
+            case FAIRY_UPGRADE_HALF_DAMAGE:
+                if (!(gSaveContext.doubleDefense)) {
+                    // Damage halved
+                    osSyncPrintf(VT_FGCOL(PURPLE) " ☆☆☆☆☆ ダメージ半減 ☆☆☆☆☆ \n" VT_RST, &gSaveContext);
+                    this->givingSpell = true;
+                    givingReward = true;
+                }
+                break;
+        }
+    }
+
+    if (givingReward) {
+        if (gSaveContext.sceneSetupIndex < 4) {
+            if (globalCtx->sceneNum != SCENE_DAIYOUSEI_IZUMI) {
+                switch (this->fountainType) {
+                    case FAIRY_SPELL_FARORES_WIND:
+                        globalCtx->csCtx.segment = SEGMENTED_TO_VIRTUAL(&D_02000160);
+                        gSaveContext.cutsceneTrigger = 1;
+                        break;
+                    case FAIRY_SPELL_DINS_FIRE:
+                        globalCtx->csCtx.segment = SEGMENTED_TO_VIRTUAL(&D_02001020);
+                        gSaveContext.cutsceneTrigger = 1;
+                        break;
+                    case FAIRY_SPELL_BLUE_DIMOND:
+                        globalCtx->csCtx.segment = SEGMENTED_TO_VIRTUAL(&D_02001F40);
+                        gSaveContext.cutsceneTrigger = 1;
+                        break;
+                }
+            } else {
+                switch (this->fountainType) {
+                    case FAIRY_UPGRADE_SPIN_ATTACK:
+                        globalCtx->csCtx.segment = SEGMENTED_TO_VIRTUAL(&D_02000130);
+                        gSaveContext.cutsceneTrigger = 1;
+                        break;
+                    case FAIRY_UPGRADE_DOUBLE_MAGIC:
+                        globalCtx->csCtx.segment = SEGMENTED_TO_VIRTUAL(&D_020013E0);
+                        gSaveContext.cutsceneTrigger = 1;
+                        break;
+                    case FAIRY_UPGRADE_HALF_DAMAGE:
+                        globalCtx->csCtx.segment = SEGMENTED_TO_VIRTUAL(&D_020025D0);
+                        gSaveContext.cutsceneTrigger = 1;
+                        break;
+                }
+            }
+        }
+        this->actionFunc = BgDyYoseizo_SetupSpinGrow_Reward;
+        return;
```

Comment:
This can be an else clause, right?

Reply:
I couldn't remember if setting an actionFunc was one of the things that we styled with returns

Reply:
theres no set rules, sometimes it makes sense to yeet out of a function early when doing a new action, sometimes it reads nice with an else. up to you

Reply:
There's one in the middle of the big function `BgDyYoseizo_Give_Reward` that would look pretty awful as an else, so I decided to be consistent in the rest of the actor